cpp/src/strings/repeat_strings.cu

```diff
@@ -166,28 +166,169 @@ std::unique_ptr<column> repeat_strings(strings_column_view const& input,
   return make_strings_column(strings_count,
                              std::move(offsets_column),
                              std::move(chars_column),
-                             input.null_count(),
-                             cudf::detail::copy_bitmask(input.parent(), stream, mr),
+                             strings.null_count(),
+                             cudf::detail::copy_bitmask(strings.parent(), stream, mr),
+                             stream,
+                             mr);
+}
+
+namespace {
+/**
+ * @brief Functor to compute string sizes and repeat the input strings, each string is repeated by a
+ * separate number of times.
+ */
+struct compute_size_and_repeat_separately_fn {
+  column_device_view const strings_dv;
+  column_device_view const repeat_times_dv;
+  bool const strings_has_nulls;
+  bool const rtimes_has_nulls;
+
+  offset_type* d_offsets{nullptr};
```

Comment:
No, the functor needs to own memory.

Reply:
That's fine by me, then. I'd like to see spans used instead, but it's not a big deal (and it might be better to make a sweeping change to keep consistency).